tests/test_e2e_integration.py

```diff
@@ -0,0 +1,241 @@
+
+        TOL = analytic_tol if dev.analytic else tol
+
+        # Skip if not logged in to Orquestra
+        try_resp = qe_list_workflow()
+        need_login_msg = "token has expired, please log in again\n"
+
+        if need_login_msg in try_resp:
+            pytest.skip("Has not logged in to the Orquestra platform.")
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.Hadamard(0)
+            return qml.expval(qml.PauliZ(0))
+
+        assert math.isclose(circuit(), 0, abs_tol=TOL)
+
+    def test_compute_expval_including_identity(self):
+        """Test a simple circuit that involves computing the expectation value of the
+        Identity operator."""
+        dev = qml.device("orquestra.qiskit", wires=3)
+
+        # Skip if not logged in to Orquestra
+        try_resp = qe_list_workflow()
+        need_login_msg = "token has expired, please log in again\n"
+
+        if need_login_msg in try_resp:
+            pytest.skip("Has not logged in to the Orquestra platform.")
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.PauliX(0)
+            qml.PauliX(1)
+            qml.PauliX(2)
+            return (
+                qml.expval(qml.Identity(0)),
+                qml.expval(qml.PauliZ(1)),
+                qml.expval(qml.Identity(2)),
+            )
+
+        assert np.allclose(circuit(), np.array([1, -1, 1]))
+
+    def test_jacobian_with_batch_execute(self):
+        """Test that the value of the jacobian computed using the internal
+        batch_execute method corresponds to the value computed with
+        the default.qubit device.
+
+        There are ``qubits * layers * 3 * 2`` many circuits to evaluate.
+        """
+        try_resp = qe_list_workflow()
+        need_login_msg = "token has expired, please log in again\n"
+
+        if need_login_msg in try_resp:
+            pytest.skip("Has not logged in to the Orquestra platform.")
+
+        qml.enable_tape()
+
+        # Evaluate 12 circuits (2 * 1 * 3 * 2)
+        # By default, this fits into two separate workflow files
+        qubits = 2
+        layers = 1
+        weights = qml.init.strong_ent_layers_uniform(layers, qubits)
+
+        dev1 = qml.device(
+            "orquestra.qiskit",
+            backend="statevector_simulator",
+            wires=qubits,
+            analytic=True,
+            keep_files=False,
+        )
+        dev2 = qml.device("default.qubit", wires=qubits, analytic=True)
+
+        def func(weights):
+            qml.templates.StronglyEntanglingLayers(weights, wires=range(qubits))
+            return qml.expval(qml.PauliZ(0))
+
+        orquestra_qnode = qml.QNode(func, dev1)
+        default_qnode = qml.QNode(func, dev2)
+
+        dfunc1 = qml.grad(orquestra_qnode)
+        dfunc2 = qml.grad(default_qnode)
+
+        res_orquestra = dfunc1(weights)
+        res_default_qubit = dfunc2(weights)
+
+        assert np.allclose(res_orquestra, res_default_qubit)
+        qml.disable_tape()
```

Comment:
@antalszava better to make a fixture for this, so that tape mode is always disabled even if the test fails middway.

```python
@pytest.fixture
def tape_mode():
    qml.enable_tape()
    yield
    qml.disable_tape()
```

Reply:
Oh nice, that's a good one! Actually just realized that these are all unit tests which construct tapes explicitly, so tape mode doesn't seem to be needed. However, changed to using this fixture in the relevant `e2e` test case.